core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt

```diff
@@ -616,398 +806,458 @@ object GraphQLActionBuilder {
+                    LongGene(tableType)
             "null" ->
-                return getReturnGene(state, tableType, kindOfTableFieldType, kindOfTableField, tableFieldType, history,
-                        isKindOfTableFieldTypeOptional, isKindOfTableFieldOptional, enumValues, methodName)
+                return getInputGene(state, tableType, kindOfTableFieldType, kindOfTableField, tableFieldType, history,
+                        isKindOfTableFieldTypeOptional, isKindOfTableFieldOptional, enumValues, methodName, unionTypes, interfaceTypes)
             "date" ->
-                return OptionalGene(tableType, DateGene(tableType))
-
+                return if (isKindOfTableFieldTypeOptional)
+                    OptionalGene(tableType, DateGene(tableType))
+                else
+                    DateGene(tableType)
             "enum" ->
-                return OptionalGene(tableType, EnumGene(tableType, enumValues))
+                return if (isKindOfTableFieldTypeOptional)
+                    OptionalGene(tableType, EnumGene(tableType, enumValues))
+                else
+                    EnumGene(tableType, enumValues)
             "scalar" ->
-                return getReturnGene(state, tableFieldType, tableType, kindOfTableFieldType, kindOfTableField, history,
-                        isKindOfTableFieldTypeOptional, isKindOfTableFieldOptional, enumValues, methodName)
+                return getInputGene(state, tableFieldType, tableType, kindOfTableFieldType, kindOfTableField, history,
+                        isKindOfTableFieldTypeOptional, isKindOfTableFieldOptional, enumValues, methodName, unionTypes, interfaceTypes)
             "id" ->
-                return OptionalGene(tableType, StringGene(tableType))
+                return if (isKindOfTableFieldTypeOptional)
+                    OptionalGene(tableType, StringGene(tableType))
+                else
+                    StringGene(tableType)
 
             "union" -> {
-                LoggingUtil.uniqueWarn(log, "Kind Of Table Field not supported yet: $kindOfTableField")
-                return OptionalGene("TODO", StringGene("TODO"))
+                LoggingUtil.uniqueWarn(log, " GQL does not support union in input type: $kindOfTableField")
+                return StringGene("Not supported type")
             }
             "interface" -> {
-                LoggingUtil.uniqueWarn(log, "Kind Of Table Field not supported yet: $kindOfTableField")
-                return OptionalGene("TODO", StringGene("TODO"))
+                LoggingUtil.uniqueWarn(log, "GQL does not support interface in input type: $kindOfTableField")
+                return StringGene("Not supported type")
             }
             else ->
-                return OptionalGene(tableType, StringGene(tableType))
+                return if (isKindOfTableFieldTypeOptional)
+                    OptionalGene(tableType, StringGene(tableType))
+                else
+                    StringGene(tableType)
+
```

Comment:
There are tree blocs of "when": two blocs for inputs and one for return. 
*For the inputs: they are different because the names are different (see eg below). 
*And for the return, it could not be refactored with inputs because we do not consider optional/non optional cases (unlike in inputs) .
eg:
```
"string" ->
    return if (isKindOfTableFieldTypeOptional)
        OptionalGene(tableType, StringGene(tableType))
    else
        StringGene(tableType)
/*****************************************************************************************/                      
"string"->
return if (isKindOfTableFieldTypeOptional)
    OptionalGene(methodName, StringGene(methodName))
else
    StringGene(methodName)
```

Reply:
add this comment to the code to clarify